pkg/m3admin/namespace/namespace.go

```diff
@@ -21,179 +21,78 @@
 package namespace
 
 import (
-	"bytes"
-	"encoding/json"
+	"errors"
 	"fmt"
-	"io/ioutil"
-	"net/url"
 
-	"github.com/m3db/m3db-operator/pkg/m3admin"
+	myspec "github.com/m3db/m3db-operator/pkg/apis/m3dboperator/v1"
 
-	namespacepb "github.com/m3db/m3/src/dbnode/generated/proto/namespace"
-	ns "github.com/m3db/m3/src/dbnode/storage/namespace"
-	nsh "github.com/m3db/m3/src/query/api/v1/handler/namespace"
+	m3ns "github.com/m3db/m3/src/dbnode/generated/proto/namespace"
 	"github.com/m3db/m3/src/query/generated/proto/admin"
-
-	"github.com/gogo/protobuf/jsonpb"
-	"go.uber.org/zap"
-)
-
-const (
-	// TODO(PS) add to spec for crd?
-	// defaults for namespace creation
-	_defaultBootstrapEnabled                                         = true
-	_defaultFlushEnabled                                             = true
-	_defaultWritesToCommitLog                                        = true
-	_defaultCleanupEnabled                                           = true
-	_defaultSnapshotEnabled                                          = false
-	_defaultRepairEnabled                                            = false
-	_defaultRententionOptionRetentionPeriodNanos                     = int64(172800000000000)
-	_defaultRententionOptionBlockSizeNanos                           = int64(7200000000000)
-	_defaultRententionOptionBufferFutureNanos                        = int64(600000000000)
-	_defaultRententionOptionBufferPastNanos                          = int64(600000000000)
-	_defaultRententionOptionBlockDataExpiry                          = true
-	_defaultRententionOptionBlockDataExpiryAfterNotAccessPeriodNanos = int64(300000000000)
-	_defaultIndexOptionenabled                                       = true
-	_defaultIndexOptionBlockSizeNanos                                = int64(7200000000000)
 )
 
-type namespaceClient struct {
-	client m3admin.Client
-	logger *zap.Logger
-	url    string
-}
-
-// Option provides an interface that can be used for setter options with the
-// constructor
-type Option interface {
-	execute(*namespaceClient) error
-}
-
-type optionFn func(n *namespaceClient) error
-
-func (fn optionFn) execute(n *namespaceClient) error {
-	return fn(n)
-}
-
-// WithURL is a setter to override the default url of m3coordinator
-func WithURL(u string) Option {
-	return optionFn(func(n *namespaceClient) error {
-		if _, err := url.ParseRequestURI(u); err != nil {
-			return err
-		}
-		n.url = u
-		return nil
-	})
-}
+// RequestFromSpec returns a namespace add request from a cluster spec namespace
+// config.
+func RequestFromSpec(ns myspec.Namespace) (*admin.NamespaceAddRequest, error) {
+	if ns.Name == "" {
```

Comment:
Is this a generally accepted way in golang to check for an empty string? I usually go with `'len(string) <= 0` but maybe that's just a java thing